### PR TITLE
Feat(Auth Test): Custom parity testing for Custom Test without SRP

### PR DESCRIPTION
**Pull request by gpanshu**
- [ ] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*

*Description of changes:*

*How did you test these changes?*
(Please add a line here how the changes were tested)

- [x] Added Unit Tests
- [ ] Added Integration Tests

*Documentation update required?*
- [x] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/SerializationTools.kt [hunk 1]
```diff
@@ -183,6 +185,15 @@ fun Any?.toJsonElement(): JsonElement {
     }
 }
 
+fun AWSCognitoAuthSignInOptions.toJsonElement(): JsonElement {
```
Comment (sktimalsina) [inline aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/SerializationTools.kt:188]: Is this necessary? gsonBased or ReflectionBased Serializers should handle these. 
Reason for having them is to have to have as less custom serializers as possible.

Comment (gpanshu) [reply aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/SerializationTools.kt]: Removed.

File: aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/JsonGenerator.kt [hunk 2]
```diff
@@ -61,6 +61,6 @@ object JsonGenerator {
 }
 
 fun main() {
-    cleanDirectory()
+    // cleanDirectory()
```
Comment (div5yesh) [inline aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/JsonGenerator.kt:64]: Do we not need to clean outdated json test files?

File: aws-auth-cognito/src/test/java/featureTest/utilities/AuthOptionsFactory.kt [hunk 3]
```diff
@@ -74,6 +76,17 @@ object AuthOptionsFactory {
         AuthAPI.getVersion -> TODO()
     } as T
 
+    private fun getSignInOptions(optionsData: JsonObject): AuthSignInOptions {
+        return if (optionsData.containsKey("signInOptions")) {
+            val authFlowType = AuthFlowType.valueOf(
+                ((optionsData["signInOptions"] as Map<String, String>)["authFlow"] as JsonPrimitive).content
+            )
+            AWSCognitoAuthSignInOptions.builder().authFlowType(authFlowType).build()
```
Comment (div5yesh) [inline aws-auth-cognito/src/test/java/featureTest/utilities/AuthOptionsFactory.kt:84]: Is metadata not required for custom auth?

**Review by div5yesh — APPROVED** [review]
Few comments, else LGTM.

File: aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/JsonGenerator.kt (hunk 2 above)
Comment (gpanshu) [reply aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/JsonGenerator.kt:64]: This step is manually run anyway so does not matter. This function also clears out the configuration files which are not regenerated so this has to be tweaked.

File: aws-auth-cognito/src/test/java/featureTest/utilities/AuthOptionsFactory.kt (hunk 3 above)
Comment (gpanshu) [reply aws-auth-cognito/src/test/java/featureTest/utilities/AuthOptionsFactory.kt:84]: metadata is not required for custom auth yes.

**Review by tylerjroach — APPROVED** [review]

**Pull request merged**